toolbox/db/db_get.m

```diff
@@ -218,6 +294,64 @@
         else
             varargout{1} = sAnatFiles;
         end
+
+%% ==== ANATOMY FILE and SURFACE FILE ====
+    % Usage: [sFiles, sItems] = db_get('AnatomyFile', FileIDs,   Fields)
+    % Usage: [sFiles, sItems] = db_get('AnatomyFile', FileNames, Fields)
+    % Usage: [sFiles, sItems] = db_get('AnatomyFile', CondQuery, Fields)
+    case {'AnatomyFile', 'SurfaceFile'}
+        % Parse inputs
+        iFiles = args{1};
+        fields = '*';                              
+        templateStruct = db_template('AnatomyFile');
+
+        if ischar(iFiles)
+            iFiles = {iFiles};
+        elseif isstruct(iFiles)
+            condQuery = args{1};           
+        end
+
+        if length(args) > 1
+            fields = args{2};
+            if ischar(fields)
+                fields = {fields};
+            end
+            for i = 1 : length(fields)
+                resultStruct.(fields{i}) = templateStruct.(fields{i});
+            end
+        else
+            resultStruct = templateStruct;
+        end
+
+        % Input is FileIDs and FileNames
+        if ~isstruct(iFiles)
+            nFiles = length(iFiles);
+            sFiles = repmat(resultStruct, 1, nFiles);
+            for i = 1:nFiles
+                if iscell(iFiles)
+                    condQuery.FileName = iFiles{i};
+                else
+                    condQuery.Id = iFiles(i);
+                end
+                sFiles(i) = sql_query(sqlConn, 'select', 'anatomyfile', fields, condQuery);
```

Comment:
Similar comment to db_get('Subject'): this will crash when the query does not return a file.

Reply:
Solved in dc223702e2fb9192576fe1f529f6e9bcd3da6c22